contracts/StablecoinConverter.sol

```diff
@@ -0,0 +1,119 @@
+pragma solidity ^0.5.0;
+
+import "./EpochTokenLocker.sol";
+import "./libraries/IdToAddressBiMap.sol";
+import "solidity-bytes-utils/contracts/BytesLib.sol";
+import "./libraries/IdToAddressBiMap.sol";
+
+
+contract StablecoinConverter is EpochTokenLocker {
+    using SafeMath for uint;
+    using BytesLib for bytes32;
+
+    event OrderPlacement(
+        address owner,
+        uint16 buyToken,
+        uint16 sellToken,
+        bool sellOrderFlag,
+        uint32 validFrom,
+        uint32 validUntil,
+        uint256 buyAmount,
+        uint256 sellAmount
+    );
+
+    event OrderCancelation(
+        address owner,
+        uint id
+    );
+
+    //outstanding volume of an order is encoded as buyAmount or sellAmount depending on sellOrderFlag
+    struct Order {
+        uint16 buyToken;
+        uint16 sellToken;
+        uint32 validFrom;  // order is valid from validFrom inclusively
+        uint32 validUntil;  // order is valid till validUntil inclusively
+        bool sellOrderFlag;
+        uint128 buyAmount;
+        uint128 sellAmount;
```

Comment:
Would it make sense to have another variable for the "remainingSellAmount" to allow to fill it partially multiple times?

Reply:
We will adapt the buyAmount and sellAmount, so that it keeps the same ratio, but reduce the amounts to indicate the lower remainingSellVolume